Remove deprecated props (#9789)
marmelab/react-admin: docs/Upgrade.md

If you were using this prop, you can safely remove it.

> ## `disableNotification` Param of `useCheckAuth` Was Removed

Reviewer: ```suggestion
## `useCheckAuth` No Longer Accepts A `disableNotification` Param
```

> The `useCheckAuth` hook no longer accepts the deprecated `disableNotification` param. To disabe the notification at the `checkAuth` call, `authProvider.checkAuth()` should return a rejected promise with a `falsy` message:

Reviewer: ```suggestion
The `useCheckAuth` hook no longer accepts the deprecated `disableNotification` param. To disable the "Authentication required" notification when calling `checkAuth`, `authProvider.checkAuth()` should return a rejected promise with the value `{ message: false }`:
```

```ts
const authProvider: AuthProvider = {
    //...
    checkAuth: () => Promise.reject({ message: false }),
}
```

> ## `disableNotification` Param of `useLogoutIfAccessDenied` Was Removed

Reviewer: ```suggestion
## `useLogoutIfAccessDenied` No Longer Accepts A `disableNotification` Param
```

> The `useLogoutIfAccessDenied` hook no longer accepts the deprecated `disableNotification` param. To disabe the notification at the `checkError` call, `authProvider.checkError()` should return a rejected promise with a `falsy` message:

Reviewer: ```suggestion
The `useLogoutIfAccessDenied` hook no longer accepts the deprecated `disableNotification` param. To disable the "Authentication required" notification when `checkError` is called, `authProvider.checkError()` should return a rejected promise with the value `{ message: false }`:
```
